packages/zk-eddsa-event-ticket-pcd/src/ZKEdDSAEventTicketPCD.ts

```diff
@@ -154,21 +154,25 @@ export async function init(args: ZKEdDSAEventTicketPCDInitArgs) {
   savedInitArgs = args;
 }
 
+async function ensureDepsInitialized(): Promise<void> {
+  if (!depsInitializedPromise) {
+    depsInitializedPromise = (async () => {
+      babyJub = await buildBabyjub();
+      eddsa = await buildEddsa();
+    })();
+  }
+
+  await depsInitializedPromise;
```

Comment:
nit: I know this was here before, but I think we can simplify to this

```suggestion
  if (!depsInitializedPromise) {
    babyJub = await buildBabyjub();
    eddsa = await buildEddsa();
  }
```

Reply:
Would that be equivalent?  Your example may need fleshing out as to where the variable used in the if statement is set.

I'm still learning how promises work from context, but I think we want to allow multiple callers to wait while a single caller does initialization (not multiple competing initializations) and the promise allows for that.  My understanding is that as soon as `await` is used other events can run, so another caller could start running its own initialization, and multiple callers could create different babyJub instances (not sure if that's safe or not, but it seems undesirable).  Is there a cleaner way to do this which preserves that property?

Reply:
Ah yes, I think the original code is right. We'll need to actually set the `depsInitializedPromise` variable to the Promise so that future callers can skip this `if` check.